GetDKAN/dkan
# Make frontend module work with Gatsby or Create React App

Create new routing system for Create React App frontend that doesn't use the same dynamic routing system as the Gatsby version. 

QA Steps:
- `dktl make --frontend --branch=react-app-module`
- `dktl install --demo` This will install the frontend module and also install the base config included with the module that is setup for Gatsby (data-catalog-frontend)
- Site should be working
- In the docroot folder of the site, `git clone https://github.com/GetDKAN/data-catalog-react frontend` to add the React app to `docroot/frontend`
- cd into the frontend app and run `yarn install && yarn build` or `npm install && npm build`
- Go to: `/admin/config/development/configuration/single/import` and import the following with the label of `frontend.config`
```
static: False
build_folder: '/build'
frontend_path: '/frontend'
routes:
  - home,/home
  - about,/about
  - api,/api
  - dataset,/dataset/{id}
  - datasetapi,/dataset/{id}/api
  - search,/search
  - publishers,/publishers 
```
- `dktl drush cr`
- Site should now be serving the React app.

If not merged yet, use `git clone --single-branch --branch add-env-files --depth 1 https://github.com/GetDKAN/data-catalog-react.git` to have .env files setup so build works correctly. The site should still load without being a dynamic React app, but the calls on the homepage, dataset, and search will fail as the axios call may be wrong.
